services/src/main/java/org/exoplatform/news/NewsServiceImpl.java

```diff
@@ -350,6 +352,18 @@ public News updateNews(News news) throws Exception {
         if ("published".equals(news.getPublicationState())) {
           publicationService.changeState(newsNode, "published", new HashMap<>());
         }
+
+        //if it's an "update news" case
+        if (StringUtils.isNotEmpty(news.getId()) && news.getCreationDate() != null) {
+          News newMentionedNews = news;
+          if (!previousMentions.isEmpty()) {
+            //clear old mentions from news body before sending a custom object to notification context.
+            previousMentions.forEach(username -> {
+              newMentionedNews.setBody(newMentionedNews.getBody().replaceAll("@"+username, ""));
+            });
+          }
+          sendNotification(newMentionedNews, NotificationConstants.NOTIFICATION_CONTEXT.MENTION_IN_NEWS);
```

Comment:
Wouldn't be better to just send the list of new mentioned users, instead of modifying et processing the news again ?